src/functions/href.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import defineFunction, {ordargument} from "../defineFunction";
+import buildCommon from "../buildCommon";
+import mathMLTree from "../mathMLTree";
+
+import * as html from "../buildHTML";
+import * as mml from "../buildMathML";
+
+defineFunction({
+    type: "href",
+    names: ["\\href"],
+    props: {
+        numArgs: 2,
+        argTypes: ["original", "original"],
```

Comment:
Should we maybe be using `"text"` for the href so that if we're in math mode and we include a URL with spaces in it, the spaces don't get eaten.

Reply:
Since #912, I don't think the space behavior will be different, but `"text"` does feel slightly more correct.  I'm guessing, in LaTeX, the argument is treated like a verbatim. For example, are two spaces preserved as two spaces or merged into one?  (In KaTeX, they will currently be merged.)  Does `\foo bar` have a space (I guess so) or remove the space (as it will probably currently behave).  This might require a new argument type of `"url"`...

Reply:
It seems that the first argument of `\href` will be treated as if it's inside verbatim, but both `\href{http://en.wikipedia.org/wiki/Equals\_sign}{=}` and `\href{http://en.wikipedia.org/wiki/Equals_sign}{=}` generate the same link to Wikipedia.

In the current implementation, it rejects `\href{http://en.wikipedia.org/wiki/Equals_sign}{=}` and accepts `\href{http://en.wikipedia.org/wiki/Equals\_sign}{=}`, but it links to `http://en.wikipedia.org/wiki/Equals/_sign` instead of `http://en.wikipedia.org/wiki/Equals_sign`. So it might be appropriate to create new argument type `url` for proper parsing, as @edemaine said.